src/math/warpExposure.cc

```diff
@@ -494,6 +495,232 @@ int warpImage(DestImageT &destImage, SrcImageT const &srcImage, geom::XYTransfor
+              WarpingControl const &control, typename DestImageT::SinglePixel padValue) {
+    if (details::isSameObject(destImage, srcImage)) {
+        throw LSST_EXCEPT(pexExcept::InvalidParameterError, "destImage is srcImage; cannot warp in place");
+    }
+    if (destImage.getBBox(image::LOCAL).isEmpty()) {
+        return 0;
+    }
+    // if src image is too small then don't try to warp
+    try {
+        std::shared_ptr<SeparableKernel> warpingKernelPtr = control.getWarpingKernel();
+        warpingKernelPtr->shrinkBBox(srcImage.getBBox(image::LOCAL));
+    } catch (...) {
+        for (int y = 0, height = destImage.getHeight(); y < height; ++y) {
+            for (typename DestImageT::x_iterator destPtr = destImage.row_begin(y), end = destImage.row_end(y);
+                 destPtr != end; ++destPtr) {
+                *destPtr = padValue;
+            }
+        }
+        return 0;
+    }
+    std::shared_ptr<SeparableKernel> warpingKernelPtr = control.getWarpingKernel();
+    int interpLength = control.getInterpLength();
+
+    std::shared_ptr<LanczosWarpingKernel const> const lanczosKernelPtr =
+            std::dynamic_pointer_cast<LanczosWarpingKernel>(warpingKernelPtr);
+
+    int numGoodPixels = 0;
+
+    // transformDestToSrc transforms from parent to parent
+    // but for warping we want local to local coordinates so make a new transform
+    std::vector<double> const destParentFromLocalVec = {static_cast<double>(destImage.getX0()),
+                                                        static_cast<double>(destImage.getY0())};
+    auto const destParentFromLocalMap = ast::ShiftMap(destParentFromLocalVec);
+    std::vector<double> const srcLocalFromParentVec = {static_cast<double>(-srcImage.getX0()),
+                                                       static_cast<double>(-srcImage.getY0())};
+    auto const srcLocalFromParentMap = ast::ShiftMap(srcLocalFromParentVec);
+    auto const localSrcFromDestMap =
+            srcLocalFromParentMap.of(srcFromDest.getFrameSet()->of(destParentFromLocalMap));
+    auto localSrcFromDest = geom::Transform<geom::Point2Endpoint, geom::Point2Endpoint>(localSrcFromDestMap);
+
+    // Get the source MaskedImage and a pixel accessor to it.
+    int const srcWidth = srcImage.getWidth();
+    int const srcHeight = srcImage.getHeight();
+    LOGL_DEBUG("TRACE2.afw.math.warp", "source image width=%d; height=%d", srcWidth, srcHeight);
+
+    int const destWidth = destImage.getWidth();
+    int const destHeight = destImage.getHeight();
+
+    LOGL_DEBUG("TRACE2.afw.math.warp", "remap image width=%d; height=%d", destWidth, destHeight);
+
+    // Set each pixel of destExposure's MaskedImage
+    LOGL_DEBUG("TRACE3.afw.math.warp", "Remapping masked image");
+
+    int const maxCol = destWidth - 1;
+    int const maxRow = destHeight - 1;
+
+    detail::WarpAtOnePoint<DestImageT, SrcImageT> warpAtOnePoint(srcImage, control, padValue);
+
+    if (interpLength > 0) {
```

Comment:
I withdraw my most recent change. It needlessly caches row information that can be computed on the fly. A clever implementation could compute the end row/column of each band and the fractional width on the fly (storing almost nothing) because all bands except the last are the same width. But that's more cleverness than I want to code.